internal/ingress/types_equals.go

```diff
@@ -479,6 +479,10 @@ func (l1 *Location) Equal(l2 *Location) bool {
 		return false
 	}
 
+	if l1.DefaultBackendUpstreamName != l2.DefaultBackendUpstreamName {
+		return false
+	}
```

Comment:
Was this the only change that fixed e2e test failure?

Reply:
This, and adding an actual deployment so that endpoints will exist (here: https://github.com/kubernetes/ingress-nginx/pull/3730/files#diff-e82a2942a511f5a12803acc64b3fc676R107)